In [7]:
import pandas as pd
import csv
df = pd.read_csv("dataset/train.csv", escapechar="\\", error_bad_lines=False, quoting=csv.QUOTE_NONE)

TITLE  \
0          Pete The Cat Bedtime Blues Doll, 14.5 Inch   
1   The New Yorker NYHM014 Refrigerator Magnet, 2 ...   
2   The Ultimate Self-Sufficiency Handbook: A Comp...   
3    Amway Nutrilite Kids Chewable Iron Tablets (100)   
4   Teacher Planner Company A4 6 Lesson Academic T...   
5   Men'S Full Sleeve Raglan T-Shirts Denim T-Shir...   
6               Glance Women's Wallet (Black) (LW-21)   
7   Wild Animals Hungry Brain Educational Flash Ca...   
8   Hindi Barakhadi Educational Flash Cards for Ki...   
9   Chapin 1380 Premier Pro 3-Gallon Tri-Poxy Stee...   
10  Slo-Niacin Polygel Controlled-Release Niacin, ...   
11  White : Braun KF400-WH Aromaster 10-Cup Coffee...   
12  Yamaha PCMSKIT Portable Electronic Keyboard Ad...   
13                   Microsoft Sidewinder Gamepad Pro   
14  Jiffy Steamer with Plastic Steam Head (Beige, ...   
15  YASHICA Open Box OPTICAL OPTICAL Zoomate 140 3...   
16  OXO Good Grips BPA-Free Turkey Baster with Cle...   
17  Sony VAIO RoomLink Network Media Receiver (PCN...   
18  Westcott 5684 5 x 6 Feet Collapsible Backdrop ...   
19  InterDesign 14 by 9 by 5-Inch Basic Basket, Me...   
20   Woodard & Charles 6.5-Inch Individual Salad Bowl   
21                 Meco Swinger Charcoal Grill, Black   
22        Crystal 3 Blue Ring Pool, 3-Ring, 66" X 16"   
23  Oscar Pour Lui By Oscar De La Renta Edt Spray ...   
24                      Farberware FPHM600 Hand Mixer   
25  3M(TM) Positionable Mounting Adhesive 568, 24 ...   
26  Zojirushi RDS-600 6-Liter Thermal Rice Warmer,...   
27            ALL GLASS AQUARIUM Pine Cabinet 24" Oak   
28                    Sabian 16 Inch HHX Studio Crash   
29  Sabian 11892XN 18-inch HHX X-TREME Crash (Golden)   
30                                 NTE 1151 IC Linear   
31  C2G/Cables to Go 27597 DVI-A Female to HD15 VG...   
32  Verbatim 8.5 GB 2.4X Double Layer Recordable D...   
33          Go Video Rave-MP AMP128 128 MB MP3 Player   
34  D'Addario EXL160TP Nickel Wound Bass Guitar St...   
35           Annemarie Borlind Eye Wrinkle Cream, 18g   
36  Golden Fluid Acrylics - Pyrrole Red - 1 oz Bottle   
37  Buddy Products Task File Organizer Rack, 4 Poc...   
38                                  Bank AmeriKid ATM   
39  Panter Company MAG-LH-RD Magnetic Label Holder...   
40  Wrangler Men's Regular Fit Jeans (13MWZPW1-463...   
41               Rio Grande Games Princes of Florence   
42                        Wilton Rolling Pin, 22.86cm   
43  Stanley National N181-461 5051 4-1/2" BOX RAIL...   
44  Oster 7 pc kit containing Oster's stiff groomi...   
45  Kong Air Dog Squeakair Tennis Balls Dog Toy - ...   
46  Sally Hansen Vitamin E Nail and Cuticle Oil, 1...   
47                                      Susan Howatch   
48                         Elizabeth Reynolds Hapgood   
49                                     Bohumil Hrabal   

                                          DESCRIPTION  \
0   Pete the Cat is the coolest, most popular cat ...   
1   The New Yorker Handsome Cello Wrapped Hard Mag...   
2                                                 NaN   
3                                                 NaN   
4                                                 NaN   
5   Men'S Full Sleeve Raglan T-Shirts Denim T-Shir...   
6   This Black wallet by Glance will be a treasure...   
7   Wild Animals are the animals that mostly stays...   
8   Hindi vowel signs and sounds are very importan...   
9   Chapin 1380 Premier Pro 3-Gallon Tri-Poxy Stee...   
10                                                NaN   
11  Color:White Product description 10 Cup Carafe ...   
12                                                NaN   
13                                                NaN   
14  Jiffy J-2000 clothes steamer, 1300 watts, with...   
15                                                NaN   
16  <b>OXO Good Grips Turkey Baster</b><br><br><ul...   
17                                                NaN   
18  Westcott offers many styles and colors of qual...   


In [8]:
df.head(50)

TITLE  \
0          Pete The Cat Bedtime Blues Doll, 14.5 Inch   
1   The New Yorker NYHM014 Refrigerator Magnet, 2 ...   
2   The Ultimate Self-Sufficiency Handbook: A Comp...   
3    Amway Nutrilite Kids Chewable Iron Tablets (100)   
4   Teacher Planner Company A4 6 Lesson Academic T...   
5   Men'S Full Sleeve Raglan T-Shirts Denim T-Shir...   
6               Glance Women's Wallet (Black) (LW-21)   
7   Wild Animals Hungry Brain Educational Flash Ca...   
8   Hindi Barakhadi Educational Flash Cards for Ki...   
9   Chapin 1380 Premier Pro 3-Gallon Tri-Poxy Stee...   
10  Slo-Niacin Polygel Controlled-Release Niacin, ...   
11  White : Braun KF400-WH Aromaster 10-Cup Coffee...   
12  Yamaha PCMSKIT Portable Electronic Keyboard Ad...   
13                   Microsoft Sidewinder Gamepad Pro   
14  Jiffy Steamer with Plastic Steam Head (Beige, ...   
15  YASHICA Open Box OPTICAL OPTICAL Zoomate 140 3...   
16  OXO Good Grips BPA-Free Turkey Baster with Cle...   
17  Sony VAIO RoomLink Network Media Receiver (PCN...   
18  Westcott 5684 5 x 6 Feet Collapsible Backdrop ...   
19  InterDesign 14 by 9 by 5-Inch Basic Basket, Me...   
20   Woodard & Charles 6.5-Inch Individual Salad Bowl   
21                 Meco Swinger Charcoal Grill, Black   
22        Crystal 3 Blue Ring Pool, 3-Ring, 66" X 16"   
23  Oscar Pour Lui By Oscar De La Renta Edt Spray ...   
24                      Farberware FPHM600 Hand Mixer   
25  3M(TM) Positionable Mounting Adhesive 568, 24 ...   
26  Zojirushi RDS-600 6-Liter Thermal Rice Warmer,...   
27            ALL GLASS AQUARIUM Pine Cabinet 24" Oak   
28                    Sabian 16 Inch HHX Studio Crash   
29  Sabian 11892XN 18-inch HHX X-TREME Crash (Golden)   
30                                 NTE 1151 IC Linear   
31  C2G/Cables to Go 27597 DVI-A Female to HD15 VG...   
32  Verbatim 8.5 GB 2.4X Double Layer Recordable D...   
33          Go Video Rave-MP AMP128 128 MB MP3 Player   
34  D'Addario EXL160TP Nickel Wound Bass Guitar St...   
35           Annemarie Borlind Eye Wrinkle Cream, 18g   
36  Golden Fluid Acrylics - Pyrrole Red - 1 oz Bottle   
37  Buddy Products Task File Organizer Rack, 4 Poc...   
38                                  Bank AmeriKid ATM   
39  Panter Company MAG-LH-RD Magnetic Label Holder...   
40  Wrangler Men's Regular Fit Jeans (13MWZPW1-463...   
41               Rio Grande Games Princes of Florence   
42                        Wilton Rolling Pin, 22.86cm   
43  Stanley National N181-461 5051 4-1/2" BOX RAIL...   
44  Oster 7 pc kit containing Oster's stiff groomi...   
45  Kong Air Dog Squeakair Tennis Balls Dog Toy - ...   
46  Sally Hansen Vitamin E Nail and Cuticle Oil, 1...   
47                                      Susan Howatch   
48                         Elizabeth Reynolds Hapgood   
49                                     Bohumil Hrabal   

                                          DESCRIPTION  \
0   Pete the Cat is the coolest, most popular cat ...   
1   The New Yorker Handsome Cello Wrapped Hard Mag...   
2                                                 NaN   
3                                                 NaN   
4                                                 NaN   
5   Men'S Full Sleeve Raglan T-Shirts Denim T-Shir...   
6   This Black wallet by Glance will be a treasure...   
7   Wild Animals are the animals that mostly stays...   
8   Hindi vowel signs and sounds are very importan...   
9   Chapin 1380 Premier Pro 3-Gallon Tri-Poxy Stee...   
10                                                NaN   
11  Color:White Product description 10 Cup Carafe ...   
12                                                NaN   
13                                                NaN   
14  Jiffy J-2000 clothes steamer, 1300 watts, with...   
15                                                NaN   
16  <b>OXO Good Grips Turkey Baster</b><br><br><ul...   
17                                                NaN   
18  Westcott offers many styles and colors of qual...   


In [10]:
df.shape

(2903024, 5)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2903024 entries, 0 to 2903023
Data columns (total 5 columns):
 #   Column          Dtype 
---  ------          ----- 
 0   TITLE           object
 1   DESCRIPTION     object
 2   BULLET_POINTS   object
 3   BRAND           object
 4   BROWSE_NODE_ID  int64 
dtypes: int64(1), object(4)
memory usage: 110.7+ MB


In [13]:
#missing values
df.isna().sum()

TITLE                 71
DESCRIPTION       723664
BULLET_POINTS     166263
BRAND              56737
BROWSE_NODE_ID         0
dtype: int64

In [23]:
pd.Series(df["TITLE"]).is_unique

False

In [24]:
df.TITLE.value_counts()

ZEBRA BLINDS Polyester Curtain for Windows Or Outdoor Décor                                                                               163
Allen Solly Men's Slim fit Casual Shirt                                                                                                   145
Candere By Kalyan Jewellers 22KT Yellow Gold Ring for Women                                                                               136
Candere By Kalyan Jewellers 18KT Yellow Gold and Diamond Ring for Women                                                                   123
Sparx Men's Pack of 2 Casual Shoes                                                                                                        110
                                                                                                                                         ... 
STUDIO NEXX Men's Light Blue Regular Fit Jeans                                                                                              1
Little

In [25]:
df.DESCRIPTION.value_counts()

Shengshou back covers are the best attractive accessory to make your mobile unique from other models .The Premium case is made out of a light, durable plastic called Polycarbonate.It is strong enough to protect it from bumps, drops and scratches.Wide array of designs and models.A raised bezel protects your display when mobile is laid flat, and the volume and power buttons are explicitly carved, so all buttons and ports are easily accessible. This case is 100% compatible with your regular charger and headphones. Kindly Note: **There may be slight change in color of design printed on case as seen on website, as there can be different monitors/tablets/smart phones settings**.                                                                                                                                                                                                                                                                                                                               

In [26]:
df.BULLET_POINTS.value_counts()

[OUT OF THE BOX DESIGNS & VIBRANT PRINT : Our collection is made by keeping in mind the rich culture and trends of our Indian country. Designs & durability of the material is top notch with high definition print quality and drop down protection.,PREMIUM QUALITY : PikkMe cases are 3D printed high quality mobile back guard,printed with high-end Korean Ink Technology.Pouches are made with premium hard Polycarbonate (plastic) material with fully matte finish for upgraded style and durability.Also they are very light weight and slim / ultra thin that perfectly fits your phone and even don’t look bulky. Note : It’s PREMIUM PLASTIC mobile back pouch and NOT silicon or rubber or fabric or leather or any other material and accessories.,ACCURATE FITTING : This cover perfectly fits your phone.These cases are made with high precision moulds covering sides and corners properly with highly accurate cut- outs for all ports and buttons.,LIFETIME WARRANTY : PikkMe provides lifetime warranty for the pri

In [27]:
df.BRAND.value_counts()

Generic                      107303
Puma                          14131
United Colors of Benetton     11821
Max                           11180
US Polo Association            8570
                              ...  
POY                               1
Kamisafe                          1
Bloomeet                          1
Sai Ram Handicrafts               1
Wham-O Holding Limited            1
Name: BRAND, Length: 258265, dtype: int64

In [28]:
df.BROWSE_NODE_ID.value_counts()

1045       215698
5           70318
1251        51929
1052        45553
4           34177
            ...  
279112          1
822060          1
1337233         1
2136811         1
248936          1
Name: BROWSE_NODE_ID, Length: 9919, dtype: int64

In [30]:
#filling the missing categorical values with the most occuring values
df = df.fillna(df.mode().iloc[0])

In [31]:
df.head()

TITLE  \
0         Pete The Cat Bedtime Blues Doll, 14.5 Inch   
1  The New Yorker NYHM014 Refrigerator Magnet, 2 ...   
2  The Ultimate Self-Sufficiency Handbook: A Comp...   
3   Amway Nutrilite Kids Chewable Iron Tablets (100)   
4  Teacher Planner Company A4 6 Lesson Academic T...   

                                         DESCRIPTION  \
0  Pete the Cat is the coolest, most popular cat ...   
1  The New Yorker Handsome Cello Wrapped Hard Mag...   
2  Shengshou back covers are the best attractive ...   
3  Shengshou back covers are the best attractive ...   
4  Shengshou back covers are the best attractive ...   

                                       BULLET_POINTS           BRAND  \
0  [Pete the Cat Bedtime Blues plush doll,Based o...     MerryMakers   
1  [Cat In A Tea Cup by New Yorker cover artist G...  The New Yorker   
2                                Skyhorse Publishing          imusti   
3  [Nutrilite Kids,Chewable Iron Tablets,Quantity...           Amway   
4  [OUT OF THE BOX DESIGNS & VIBRANT PRINT : Our ...         Generic   

   BROWSE_NODE_ID  
0               0  
1               1  
2               2  
3               3  
4               4

In [32]:
df.isna().sum()

TITLE             0
DESCRIPTION       0
BULLET_POINTS     0
BRAND             0
BROWSE_NODE_ID    0
dtype: int64

In [33]:
df.BROWSE_NODE_ID.value_counts()

1045       215698
5           70318
1251        51929
1052        45553
4           34177
            ...  
279112          1
822060          1
1337233         1
2136811         1
248936          1
Name: BROWSE_NODE_ID, Length: 9919, dtype: int64

We can drop those rows from our dataset which has the least value counts for example in BROWSE_NODE_ID there are some values like 279112, 822060 which occurs only once so lets drop these rows which occurs only once